clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -1023,9 +1023,14 @@ protected void close(Timer timer) {
     }
 
     /**
+     * Leaving the group. This method also sends LeaveGroupRequest and log {@code leaveReason} if this is dynamic members
+     * or unknown coordinator or state is not UNJOINED or this generation has a valid member id.
```

Comment:
I think it may be more useful to describe the cases where it will _not_ send a LeaveGroup and describe what this actually means (also it should have been 'and' not 'or' in the original):
```suggestion
     * Sends LeaveGroupRequest and logs the {@code leaveReason}, unless this member is using 
     * static  membership or is already not part of the group (ie does not have a valid member id, 
     * is in the UNJOINED state, or the coordinator is unknown).
```